framework/src/wifi_manager/wifi_manager_state.c

```diff
@@ -545,6 +557,7 @@ wifi_manager_result_e _handler_on_softap_state(wifimgr_msg_s *msg)
 		WIFIMGR_CHECK_RESULT(_wifimgr_stop_softap(), (TAG, "critical error\n"), WIFI_MANAGER_FAIL);
 		WIFIMGR_CHECK_RESULT(_wifimgr_run_sta(), (TAG, "critical error\n"), WIFI_MANAGER_FAIL);
 		WIFIMGR_SEND_API_SIGNAL(msg->signal);
+		/* There is an issue here, if STA is currently connected, the state set will be incorrect */
```

Comment:
Hi @a-jhanwar , I think we might need to have new states in TizenRT's wifi manager to handle concurrent mode (Whether STA is connected or disconnected while SoftAP is running).
For example, there is a state issue in the current implementation. If we try to stop SoftAP while STA is connected, the state will be set incorrectly to STA_DISCONNECTED. What do you think?
Thanks

Reply:
Thanks for the PRs.
Yes, indeed there is changes required in wifi manager to handle concurrent mode.
Detailed end user application use case is still not finalized yet.
Which may/ may not affect the changes in state of wifi manager.
Currently, I am focusing to make Bridge mode working for at least following scenario.
1. Join network on wlan0
2. start softap on wlan1
3. Device connected on wlan1, should access internet.
wm_test start
wm_test join SSID1 wpa2_aes 12345678
wm_test softap SSID2 12345678 3

On top of Last PR from RTK, we find some issue in our code.
Fixing those. 
Once this basic scenario is working (if require some temporary change in wifi manager states). Then we can start remaining changes for wifi manager new State handling, and then we can continue on PR#3 and PR#4.

I will commit remaining changes in 1-2 days to this and inform you about the status.